### PR TITLE
refactor: rename CollectorOptionsForm and add attachedServiceAccountType state to collector-form-store

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description

SSIA

### CODE REVIEW

File: apps/web/src/services/asset-inventory/collector/shared/collector-forms/collector-form-store.ts [hunk 1]
```diff
@@ -20,8 +22,9 @@ export const useCollectorFormStore = defineStore('collector-form', {
         autoUpgrade: false,
         scheduleHours: [] as number[],
         schedulePower: false,
-        attachedServiceAccount: null as AttachedServiceAccount|null,
-        pluginMetadata: {} as Record<string, any>,
+        attachedServiceAccount: [] as AttachedServiceAccount,
+        attachedServiceAccountType: 'all' as AttachedServiceAccountType,
```
Comment (sulmoJ) [inline apps/web/src/services/asset-inventory/collector/shared/collector-forms/collector-form-store.ts:26]: I love it

File: apps/web/src/services/asset-inventory/collector/shared/collector-forms/collector-form-store.ts [hunk 2]
```diff
@@ -81,12 +84,15 @@ export const useCollectorFormStore = defineStore('collector-form', {
...
             if (!hoursOnly) this.schedulePower = this.originCollector?.schedule?.state === 'ENABLED' ?? false;
         },
-        setAttachedServiceAccount(serviceAccount: AttachedServiceAccount|null) {
-            if (!serviceAccount?.length) this.attachedServiceAccount = null;
-            else this.attachedServiceAccount = serviceAccount;
+        resetAttachedServiceAccount() {
+            this.attachedServiceAccount = this.originCollector?.secret_filter?.service_accounts ?? [];
+            this.attachedServiceAccountType = this.originCollector?.secret_filter?.state === 'ENABLED' ? 'specific' : 'all';
+        },
+        setOptions(options: CollectorOptions) {
```
Comment (sulmoJ) [inline apps/web/src/services/asset-inventory/collector/shared/collector-forms/collector-form-store.ts:91]: Nice Type

**Review by sulmoJ — APPROVED** [review]
LGTM!!!

File: apps/web/src/services/asset-inventory/collector/shared/collector-forms/AttachedServiceAccountForm.vue [hunk 3]
```diff
@@ -142,34 +144,38 @@ const {
...
+        attachedServiceAccount: [],
+    });
 };
 
 const handleSelectAttachedServiceAccount = (selectedValue: string[]) => {
     setForm('selectedAttachedServiceAccount', selectedValue);
-    collectorFormStore.setAttachedServiceAccount(selectedValue);
+    collectorFormStore.$patch({
+        attachedServiceAccount: selectedValue,
+    });
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/shared/collector-forms/AttachedServiceAccountForm.vue:168]: 😎

**Review by yuda110 — APPROVED** [review]
LGTM!

**Pull request merged**